<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/pandanuc/albert_ws/src/albert/albert_description/urdf/albert.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="albert_description">
  <material name="Boxer_Black">
    <color rgba="0.1 0.1 0.1 1.0"/>
  </material>
  <material name="Boxer_Blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <material name="Boxer_Purple">
    <color rgba="0.6 0.0 0.6 1.0"/>
  </material>
  <material name="Boxer_Green">
    <color rgba="0.0 0.8 0.0 1.0"/>
  </material>
  <material name="Boxer_LimeGreen">
    <color rgba="0.3 0.8 0.3 1.0"/>
  </material>
  <material name="Boxer_Grey">
    <color rgba="0.5 0.5 0.5 1.0"/>
  </material>
  <material name="Boxer_DarkGrey">
    <color rgba="0.3 0.3 0.3 1.0"/>
  </material>
  <material name="Boxer_Red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="Boxer_Pink">
    <color rgba="0.8 0.4 0.8 1.0"/>
  </material>
  <material name="Boxer_White">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <material name="Boxer_Yellow">
    <color rgba="0.8 0.8 0.0 1.0"/>
  </material>
  <material name="Boxer_GreyTransparent">
    <color rgba="0.5 0.5 0.5 0.5"/>
  </material>
  <link name="world"/>
  <joint name="panda_joint_world" type="fixed">
    <parent link="world"/>
    <child link="base_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <link name="base_link"/>
   <joint name="mobile_joint_x" type="prismatic">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <parent link="base_link"/>
    <child link="panda_link0"/>
    <origin rpy="0 0 0" xyz="0 0 0.00"/>
    <axis xyz="1 0 0"/>
    <limit effort="87" lower="-5" upper="5" velocity="2.1750"/>
  </joint>
  <link name="panda_link0">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link0.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 1.5707963267948966 0" xyz="-0.075 0 0.06"/>
      <geometry>
        <cylinder length="0.03" radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="-0.06 0 0.06"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="-0.09 0 0.06"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
  </link>
  <link name="panda_link1">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link1.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.1915"/>
      <geometry>
        <cylinder length="0.2830" radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.333"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.05"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_joint1" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="0 0 0" xyz="0 0 0.333"/>
    <parent link="panda_link0"/>
    <child link="panda_link1"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-2.8973" upper="2.8973" velocity="2.1750"/>
  </joint>
  <link name="panda_link2">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link2.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.12" radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.06"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.06"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_joint2" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-1.7628" soft_upper_limit="1.7628"/>
    <origin rpy="-1.5707963267948966 0 0" xyz="0 0 0"/>
    <parent link="panda_link1"/>
    <child link="panda_link2"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-1.7628" upper="1.7628" velocity="2.1750"/>
  </joint>
  <link name="panda_link3">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link3.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.145"/>
      <geometry>
        <cylinder length="0.15" radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.22"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.07"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_joint3" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="1.5707963267948966 0 0" xyz="0 -0.316 0"/>
    <parent link="panda_link2"/>
    <child link="panda_link3"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-2.8973" upper="2.8973" velocity="2.1750"/>
  </joint>
  <link name="panda_link4">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link4.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.12" radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.06"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.06"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_joint4" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-3.0718" soft_upper_limit="-0.0698"/>
    <origin rpy="1.5707963267948966 0 0" xyz="0.0825 0 0"/>
    <parent link="panda_link3"/>
    <child link="panda_link4"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-3.0718" upper="-0.0698" velocity="2.1750"/>
  </joint>
  <link name="panda_link5">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link5.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.26"/>
      <geometry>
        <cylinder length="0.1" radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.31"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.21"/>
      <geometry>
        <sphere radius="0.09"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0.08 -0.13"/>
      <geometry>
        <cylinder length="0.14" radius="0.055"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0.08 -0.06"/>
      <geometry>
        <sphere radius="0.055"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0.08 -0.20"/>
      <geometry>
        <sphere radius="0.055"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_joint5" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="-1.5707963267948966 0 0" xyz="-0.0825 0.384 0"/>
    <parent link="panda_link4"/>
    <child link="panda_link5"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-2.8973" upper="2.8973" velocity="2.6100"/>
  </joint>
  <link name="panda_link6">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link6.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.03"/>
      <geometry>
        <cylinder length="0.08" radius="0.08"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.01"/>
      <geometry>
        <sphere radius="0.08"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.07"/>
      <geometry>
        <sphere radius="0.08"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_joint6" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-0.0175" soft_upper_limit="3.7525"/>
    <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
    <parent link="panda_link5"/>
    <child link="panda_link6"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-0.0175" upper="3.7525" velocity="2.6100"/>
  </joint>
  <link name="panda_link5_offset">
    </link>
  <joint name="panda_joint5_offset" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 -0.20"/>
    <parent link="panda_link5"/>
    <child link="panda_link5_offset"/>
  </joint>
  <link name="panda_link7">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link7.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.01"/>
      <geometry>
        <cylinder length="0.14" radius="0.07"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.08"/>
      <geometry>
        <sphere radius="0.07"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.06"/>
      <geometry>
        <sphere radius="0.07"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_joint7" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8973" soft_upper_limit="2.8973"/>
    <origin rpy="1.5707963267948966 0 0" xyz="0.088 0 0"/>
    <parent link="panda_link6"/>
    <child link="panda_link7"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-2.8973" upper="2.8973" velocity="2.6100"/>
  </joint>
  <link name="panda_link8">
    <collision>
      <origin rpy="3.141592653589793 1.5707963267948966 1.5707963267948966" xyz="0.0424 0.0424 -0.0250"/>
      <geometry>
        <cylinder length="0.01" radius="0.06"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0.0424 0.0424 -0.02"/>
      <geometry>
        <sphere radius="0.06"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0.0424 0.0424 -0.03"/>
      <geometry>
        <sphere radius="0.06"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_joint8" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.107"/>
    <parent link="panda_link7"/>
    <child link="panda_link8"/>
  </joint>
  <joint name="panda_hand_joint" type="fixed">
    <parent link="panda_link8"/>
    <child link="panda_hand"/>
    <origin rpy="0 0 2.356194490192345" xyz="0 0 0"/>
  </joint>
  <material name="white">
    <color rgba="0.8 0.8 0.8 1.0"/>
  </material>
  <!-- ee link -->
  <link name="panda_hand">
    <inertial>
      <mass value="0.73"/>
      <origin xyz="-0.01 0 0.03"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.0025" iyz="0" izz="0.0017"/>
    </inertial>
  </link>
  <link name="panda_vacuum_support_link">
    <inertial>
      <mass value="0.00"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.33" radius="0.01"/>
      </geometry>
      <material name="panda_vacuum_support">
        <color rgba="0.7 0.7 0.7 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.33" radius="0.01"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_vacuum_support_joint" type="fixed">
    <parent link="panda_hand"/>
    <child link="panda_vacuum_support_link"/>
    <origin rpy="0 0 0" xyz="0 0.0 0.185"/>
  </joint>
  <link name="panda_vacuum1_link">
    <inertial>
      <mass value="0.00"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.02" radius="0.02"/>
      </geometry>
      <material name="vacuum">
        <color rgba="0 0 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.02" radius="0.02"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_vacuum1_joint" type="fixed">
    <parent link="panda_hand"/>
    <child link="panda_vacuum1_link"/>
    <origin rpy="0 0 0" xyz="0.0275 0 0.33"/>
  </joint>
  <link name="panda_vacuum2_link">
    <inertial>
      <mass value="0.00"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.02" radius="0.02"/>
      </geometry>
      <material name="vacuum">
        <color rgba="0 0 1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.02" radius="0.02"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda_vacuum2_joint" type="fixed">
    <parent link="panda_hand"/>
    <child link="panda_vacuum2_link"/>
    <origin rpy="0 0 0" xyz="-0.0275 0 0.33"/>
  </joint>
  <link name="panda_rightfinger">
    <inertial>
      <mass value="0.015"/>
      <inertia ixx="2.3749999999999997e-06" ixy="0" ixz="0" iyy="2.3749999999999997e-06" iyz="0" izz="7.5e-07"/>
    </inertial>
  </link>
  <link name="panda_leftfinger">
    <inertial>
      <mass value="0.015"/>
      <inertia ixx="2.3749999999999997e-06" ixy="0" ixz="0" iyy="2.3749999999999997e-06" iyz="0" izz="7.5e-07"/>
    </inertial>
  </link>
  <gazebo reference="panda_leftfinger">
    <collision>
      <max_contacts>10</max_contacts>
      <surface>
        <contact>
          <ode>
            <!-- These two parameter need application specific tuning. -->
            <!-- Usually you want no "snap out" velocity and a generous -->
            <!-- penetration depth to keep the grasping stable -->
            <max_vel>0</max_vel>
            <min_depth>0.001</min_depth>
          </ode>
        </contact>
        <friction>
          <ode>
            <mu>1.13</mu>
            <mu2>1.13</mu2>
          </ode>
        </friction>
        <bounce/>
      </surface>
    </collision>
  </gazebo>
  <gazebo reference="panda_rightfinger">
    <collision>
      <max_contacts>10</max_contacts>
      <surface>
        <contact>
          <ode>
            <!-- These two parameter need application specific tuning. -->
            <!-- Usually you want no "snap out" velocity and a generous -->
            <!-- penetration depth to keep the grasping stable -->
            <max_vel>0</max_vel>
            <min_depth>0.001</min_depth>
          </ode>
        </contact>
        <friction>
          <ode>
            <mu>1.13</mu>
            <mu2>1.13</mu2>
          </ode>
        </friction>
        <bounce/>
      </surface>
    </collision>
  </gazebo>
  <joint name="panda_finger_joint1" type="prismatic">
    <parent link="panda_hand"/>
    <child link="panda_leftfinger"/>
    <origin rpy="0 0 0" xyz="0 0 0.0584"/>
    <axis xyz="0 1 0"/>
    <limit effort="100" lower="0.0" upper="0.04" velocity="0.2"/>
    <dynamics damping="0.03" friction="0.0"/>
  </joint>
  <joint name="panda_finger_joint2" type="prismatic">
    <parent link="panda_hand"/>
    <child link="panda_rightfinger"/>
    <origin rpy="0 0 0" xyz="0 0 0.0584"/>
    <axis xyz="0 -1 0"/>
    <limit effort="100" lower="0.0" upper="0.04" velocity="0.2"/>
    <dynamics damping="0.03" friction="0.0"/>
    <mimic joint="panda_finger_joint1"/>
  </joint>
  <material name="aluminum">
    <color rgba="0.5 0.5 0.5 1"/>
  </material>
  <material name="plastic">
    <color rgba="0.1 0.1 0.1 1"/>
  </material>
  <!-- <xacro:include filename="$(find realsense2_description)/urdf/_usb_plug.urdf.xacro" /> -->
  <!-- new root link because we calibrated w.r.t. this -->
  <joint name="camera_calibration" type="fixed">
    <origin rpy="-0.08220003 -0.01235345  2.33171179" xyz="0.1052022 0.04861002 0.02188261"/>
    <parent link="panda_link8"/>
    <child link="camera_color_optical_frame"/>
  </joint>
  <link name="camera_color_optical_frame"/>
  <joint name="camera_color_optical_joint" type="fixed">
    <origin rpy="1.5707963267948966 -1.5707963267948966 0" xyz="0.015 0 0"/>
    <parent link="camera_color_optical_frame"/>
    <child link="camera_link"/>
  </joint>
  <link name="camera_link">
    <visual>
      <!-- the mesh origin is at front plate in between the two infrared camera axes -->
      <origin rpy="1.5707963267948966 0 1.5707963267948966" xyz="0.0043 -0.0175 0"/>
      <geometry>
        <mesh filename="package://realsense2_description/meshes/d435.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 -0.0175 0"/>
      <geometry>
        <box size="0.02505 0.09 0.025"/>
      </geometry>
    </collision>
    <!-- <inertial> -->
    <!-- The following are not reliable values, and should not be used for modeling -->
    <!-- <mass value="0.076" /> -->
    <!-- <origin xyz="0 0 0" /> -->
    <!-- <inertia ixx="0.003881243" ixy="0.0" ixz="0.0" iyy="0.000498940" iyz="0.0" izz="0.003879257" /> -->
    <!-- </inertial> -->
  </link>
  <!-- camera depth joints and links -->
  <joint name="camera_depth_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="camera_link"/>
    <child link="camera_depth_frame"/>
  </joint>
  <link name="camera_depth_frame"/>
  <joint name="camera_depth_optical_joint" type="fixed">
    <origin rpy="-1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
    <parent link="camera_depth_frame"/>
    <child link="camera_depth_optical_frame"/>
  </joint>
  <link name="camera_depth_optical_frame"/>
  <!-- camera left IR joints and links -->
  <joint name="camera_infra1_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0.0 0"/>
    <parent link="camera_link"/>
    <child link="camera_infra1_frame"/>
  </joint>
  <link name="camera_infra1_frame"/>
  <joint name="camera_infra1_optical_joint" type="fixed">
    <origin rpy="-1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
    <parent link="camera_infra1_frame"/>
    <child link="camera_infra1_optical_frame"/>
  </joint>
  <link name="camera_infra1_optical_frame"/>
  <!-- camera right IR joints and links -->
  <joint name="camera_infra2_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 -0.05 0"/>
    <parent link="camera_link"/>
    <child link="camera_infra2_frame"/>
  </joint>
  <link name="camera_infra2_frame"/>
  <joint name="camera_infra2_optical_joint" type="fixed">
    <origin rpy="-1.5707963267948966 0 -1.5707963267948966" xyz="0 0 0"/>
    <parent link="camera_infra2_frame"/>
    <child link="camera_infra2_optical_frame"/>
  </joint>
  <link name="camera_infra2_optical_frame"/>
  <!-- <xacro:if value="${add_plug}"> -->
  <!--   <xacro:usb_plug parent="${name}_link" name="${name}_usb_plug"> -->
  <!--     <origin xyz="${d435_cam_mount_from_center_offset - 0.02095} ${-d435_cam_depth_py - 0.0353} 0" rpy="0 0 0"/> -->
  <!--   </xacro:usb_plug> -->
  <!-- </xacro:if> -->
</robot>


